File: PredictTheWinner.cpp
class Solution {
public:
    int winner(vector<int>& nums, int sInd, int eInd){
        return sInd == eInd ?
                nums[sInd] :
                max(nums[sInd] - winner(nums, sInd + 1, eInd), nums[eInd] - winner(nums, sInd, eInd - 1));
    }
    bool PredictTheWinner(vector<int>& nums) {
        return winner(nums, 0, nums.size() - 1) >= 0;
    }
    
};

File: PredictTheWinner.cs
public class Solution {
    int Rec(int [] nums, int sINd, int eInd)
    {
        return sINd == eInd ? nums[sINd] : Math.Max(nums[sINd] - Rec(nums, sINd + 1, eInd), nums[eInd] - Rec(nums, sINd, eInd - 1));
    }
    public bool PredictTheWinner(int[] nums)
    {
        return Rec(nums, 0, nums.Length - 1) >= 0;
    }
}
